core/templates/dev/head/pages/exploration_player/PlayerTranscriptService.js

```diff
@@ -40,22 +40,15 @@ oppia.factory('PlayerTranscriptService', [
         transcript = [];
         numAnswersSubmitted = 0;
       },
-      getStateHistory: function() {
-        var result = [];
+      hasEncounteredStateBefore: function(stateName) {
+        var stateHistory = [];
```

Comment:
You don't even need to make a list, right? Just do something like this:

    return transcript.some(function(transcriptItem) {
       return transcriptItem.getStateName() === stateName;
    });

Reply:
Yeah, done.